### PR TITLE
Customize MCUboot for rideable

### PR DESCRIPTION
**Pull request by gnomeadrift**
Just a few changes to expose some library functions for app-side use.

Also looks like I was a few commits ahead by using runtimeco's version to begin with.

### CODE REVIEW

File: boot/boot_serial/src/boot_serial.c [hunk 1]
```diff
@@ -55,6 +55,10 @@
 #include "boot_serial/boot_serial.h"
 #include "boot_serial_priv.h"
 
+#ifdef CONFIG_BOOT_ERASE_PROGRESSIVELY
```
Comment (schultetwin) [inline boot/boot_serial/src/boot_serial.c:58]: Some of these changes are bringing us up to date with a version of master. I'd prefer to stay with mcuboot releases unless we need some feature for upstream?

File: boot/bootutil/include/bootutil/bootutil.h [hunk 2]
```diff
@@ -75,6 +75,15 @@ struct image_trailer {
     uint8_t magic[16];
 };
 
+//Return error codes from boot_go
+#define BOOT_EFLASH     1
```
Comment (schultetwin) [inline boot/bootutil/include/bootutil/bootutil.h:79]: These should be removed from boot/bootutil/src/bootutil_priv.h if they're made public here. But I agree these should be public.

**Review by schultetwin — APPROVED** [review]
Looks fine to me.

**Comment by gnomeadrift** [comment]
@schultetwin I don't have access to merge this.  Can you merge it?

**Pull request merged**